tests/unit/components/status/HeatMapColorBar.spec.js

```diff
@@ -0,0 +1,278 @@
+import { renderToString } from "@vue/server-test-utils";
+import { mount } from "@vue/test-utils";
+import { shallowMount } from "@vue/test-utils";
+import ComponentToTest from "@/components/status/HeatMapColorBar.vue";
+import { HeatMapColorBar as DistComponentToTest } from "@/dist/mantarray.common";
+
+import { createLocalVue } from "@vue/test-utils";
+
+let wrapper = null;
+
+const localVue = createLocalVue();
+
+describe("HeatMapColorBar.vue", () => {
+  // afterEach(() => wrapper.destroy());
+  test("When mounting HeatMapColorBar from the build dist file, Then it loads successfully and upper_range text '100 μN' is rendered as defined", () => {
+    const propsData = {
+      upper_range: "100",
+    };
+    wrapper = shallowMount(DistComponentToTest, {
+      propsData,
+      localVue,
+    });
+    const target_span = wrapper.find(".span__heatmap-scale-higher-value");
+    expect(target_span.text()).toStrictEqual("100 μN");
+  });
+  test("Given that the uuid, upper_range, lower_range and height, When mounted successfully, Then assert that upper_range,lower_range,height are applied and as uuid is match a default linear-gradient is applied on HeatMapColorBar", async () => {
```

Comment:
I don't know what you mean by this. This is not a full thought / grammatical English `Given that the uuid, upper_range, lower_range and height`.  Given what? Nothing is being described about uuid or height etc.

`as the uuid is a match` is something about the way the test is executed, not about the result. That should be in either the `Given` or `When` section. `Then` should only contain the information about what is being asserted.

If the test is about what happens during a mount, there doesn't need to be a `Given` section.  Something like: `When mounted with a UUID matching a pre-defined color set, an upper range, a lower range, and a height, Then the text for maximum and minimum values match the props, the height of the component matches the prop, and the color gradient matches the pre-defined values for the UUID from the prop`

Reply:
As the JEST and Vue-test-utils doesn't provide the essential asserting and reactive feature of the component. The question is for web artifacts, there are no testing required as per the general opinion and its a redundant testcase

Reply:
Your reply does not make sense here.  This is a not a comment about testing reactivity, it's a comment about making sure test titles are clear and descriptive